system/Router/Router.php

```diff
@@ -573,32 +578,45 @@ public function autoRoute(string $uri)
 	//--------------------------------------------------------------------
 
 	/**
-	 * Attempts to validate the URI request and determine the controller path.
+	 * Scans the controller directory, attempting to locate a controller matching the supplied uri $segments
 	 *
 	 * @param array $segments URI segments
 	 *
-	 * @return array URI segments
+	 * @return array returns an array of remaining uri segments that don't map onto a directory
 	 */
-	protected function validateRequest(array $segments): array
+	protected function scanControllers(array $segments): array
```

Comment:
As this method is not flagged as `@internal`, changing the name is technically BC breaking.

Reply:
> As this method is not flagged as `@internal`, changing the name is technically BC breaking.

In order for this change to break any code, someone would have to have extended the router class. I did a grep search of the source and could find no place where it extends this class. This returns no results.
```
grep -r "extends.*Router" CodeIgniter4
```
If we are really concerned about it being a breaking change, perhaps we can define `validateRequest()` to simply call `scanControllers` and deprecate it:
```php
/**
 * @deprecated
 */
protected function validateRequest(array $segments): array
{
    return $this->scanControllers($segments);
}
```

If you want me to change the function name back to `validateRequest` I will do it, but it is a really bad name that does NOT describe what that function is doing.

Reply:
But users may be extending the Router class for their own use case and we cannot be sure that they did not use this method. I'll go with the deprecation instead and reroute calls to scanController.